### PR TITLE
add show feature to the user

### PR DESCRIPTION
**Pull request by Alaatohamy**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class UsersController < ApplicationController
+
```
Comment (dinahamamo) [inline app/controllers/users_controller.rb]: before_action :set_user

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  # has_many: cars, dependent: :destroy
```
Comment (dinahamamo) [inline app/models/user.rb]: has_many :cars

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  # has_many: hh_stop_points
```
Comment (dinahamamo) [inline app/models/user.rb]: has_many :hh_stop_points

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  # has_many:location_requests, dependent: :destroy
```
Comment (dinahamamo) [inline app/models/user.rb]: has_many :location_requests

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  # has_many: trips, dependent: :destroy
```
Comment (dinahamamo) [inline app/models/user.rb]: has_many :trips

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+  validates :first_name, :last_name, :email, :phone, presence: true
```
Comment (dinahamamo) [inline app/models/user.rb]: add :hashed_password, :rate, :points, :gender, presence: true

File: app/views/users/show.json.jbuilder [hunk 7]
```diff
@@ -0,0 +1,9 @@
+json.extract! @user, :first_name, :last_name, :email, :phone, :profile_pic, :rate, :points
```
Comment (dinahamamo) [inline app/views/users/show.json.jbuilder]: json.extract!(@user, :first_name, :last_name, :email, :phone, :profile_pic, :rate, :points

File: app/views/users/show.json.jbuilder [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+
+  # json.cars do
+  #   json.array! @user.cars do |car|
+  #     json.model car.model
+  #     json.color car.color
+  #     json.number car.number
+  #   end
```
Comment (dinahamamo) [inline app/views/users/show.json.jbuilder:8]: Add json.trips as well

File: db/migrate/20171224151221_create_users.rb [hunk 9]
```diff
@@ -0,0 +1,17 @@
+class CreateUsers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :users do |t|
+      t.string :first_name, null: false
+      t.string :last_name, null: false
+      t.string :email, null: false
+      t.string :hashed_password
```
Comment (dinahamamo) [inline db/migrate/20171224151221_create_users.rb:7]: null: false

File: db/migrate/20171224151221_create_users.rb [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+      t.string :phone, null: false
+      t.string :profile_pic
+      t.boolean :gender
+      t.integer :rate
```
Comment (dinahamamo) [inline db/migrate/20171224151221_create_users.rb]: default value is 0

File: db/migrate/20171224151221_create_users.rb [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+      t.integer :points
```
Comment (dinahamamo) [inline db/migrate/20171224151221_create_users.rb]: default value is 100

File: db/migrate/20171224151221_create_users.rb (hunk 9 above)
Comment (Alaatohamy) [reply db/migrate/20171224151221_create_users.rb:7]: !!

**Pull request merged**